admin_guide/http_proxies.adoc

```diff
@@ -0,0 +1,167 @@
+access, include the `*NO_PROXY*` parameter with the registry's hostname, the
+registry service's IP address, and service name. This blacklists that registry's
+address, leaving the external http proxy as the only option.
+
+. Edit the *_/etc/sysconfig/docker_* file and add the variables in shell format:
++
+====
+----
+HTTP_PROXY=http://USERNAME:PASSWORD@10.0.1.1:8080/
+HTTPS_PROXY=https://USERNAME:PASSWORD@10.0.0.1:8080/
+NO_PROXY=master.hostname.example.com,172.30.123.45,docker-registry.default.svc.cluster.local
+----
+====
+
+. Restart the Docker service:
++
+----
+# systemctl restart docker
+----
+
+////
+. If you had already imported ImageStreams before creating the proxy
+configuration, delete, then re-run the process:
++
+----
+# oc delete imagestreams -n openshift --all
+# oc create -f image-streams.json -n openshift
+----
+////
+
+[[s2i-builds]]
+
+== S2I Builds
+
+S2I builds fetch dependencies from various locations. You can
+link:../dev_guide/builds.html#environment-files[use a *_.sti/environment_* file]
+to specify simple shell variables and OpenShift will react accordingly when
+seeing build images.
+
+The following are the supported proxy environment variables with example values:
+
+====
+----
+HTTP_PROXY=http://USERNAME:PASSWORD@10.0.1.1:8080/
+HTTPS_PROXY=https://USERNAME:PASSWORD@10.0.0.1:8080/
+NO_PROXY=master.hostname.example.com
+----
+====
+
+[[configuring-default-templates-for-proxies]]
+
+== Configuring Default Templates for Proxies
+
+The example templates available in OpenShift by default do not include settings
+for HTTP proxies. For existing applications based on these templates, modify the
+`*source*` section of the buildConfig and add proxy settings:
+
+====
+"source": {
+  "type": "Git",
+  "git": {
+    "uri": "https://github.com/openshift/ruby-hello-world.git",
+    "httpProxy"  : "http://proxy.example.com",
+    "httpsProxy" : "https://proxy.example.com"
+  }
+}
+====
+
+This is similar to the process for link:#git-repository-access[using proxies for
+Git cloning].
+
+[[setting-environment-variables-in-pods]]
+
+== Setting Environment Variables in Pods
+
+Set the following environment variables in a DeploymentConfig to pass database
+connection information. The same can be done for configuring a Pod's proxy at
+runtime:
+
+====
+----
+{
+  "apiVersion": "v1beta1",
+  "kind": "DeploymentConfig",
+  "metadata": {
+    "name": "frontend"
+  },
+  "template": {
+    "controllerTemplate": {
+      "podTemplate": {
+        "desiredState": {
+          "manifest": {
+            "containers": [
+              {
+                "env": [
+                  {
+                    "name": "HTTP_PROXY",
+                    "value": "http://USER:PASSWORD@IPADDR:PORT"
+                  },
+----
+====
+
+You can also use the `oc env` command to update a DeploymentConfig with a new
+environment variable:
+
+----
+oc env <object_type>/<object_type_name> <EN_VAR>=/<VALUE>
+----
```

Comment:
I feel like it may be more helpful/appropriate here to just show an actual example cmd w/ the required env var already in place, versus just showing the generic syntax, and maybe link to the dev_guide/envars topic in-line if they wanted more info on the cmd itself. Again, we're just showing what env var needs to be set, AFAICT. E.g.:

```
====
----
oc env dc/frontend HTTP_PROXY=http://USER:PASSWORD@IPADDR:PORT
----
====